packages/firebase_snippets_app/lib/snippets/auth.dart

```diff
@@ -0,0 +1,622 @@
+// ignore_for_file: non_constant_identifier_names, avoid_print
+
+import 'dart:convert';
+import 'dart:math';
+
+import 'package:crypto/crypto.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:firebase_snippets_app/snippets/snippet_base.dart';
+import 'package:flutter/foundation.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+import 'package:sign_in_with_apple/sign_in_with_apple.dart';
+import 'package:twitter_login/twitter_login.dart';
+
+class AuthenticationSnippets implements DocSnippet {
+  @override
+  void runAll() {
+    getStarted_authStateChanges();
+    getStarted_idTokenChanges();
+    getStarted_userChanges();
+    manageUsers_getProfile();
+    passwordAuth_createPasswordBasedAccount();
+    emailLinkAuth_sendAuthLinkEmail();
+    social_google_sign_in();
+    social_twitterAuth();
+    social_facebookAuth();
+    phoneAuth_verifyPhoneNumber();
+    customAuth_withFirebase();
+    anonymousAuth_withFirebase();
+  }
+
+  void getStarted_authStateChanges() async {
+    // [START get_started_auth_state_changes]
+    FirebaseAuth.instance.authStateChanges().listen((User? user) {
+      if (user == null) {
+        print('User is currently signed out!');
```

Comment:
I like that the corresponding [web snippet](https://firebase.google.com/docs/auth/web/manage-users#get_the_currently_signed-in_user) links to the `User` object  Can you do that here too?

Also can you add ellipses comments (`// ...`) in both blocks to indicate that we expect them to write code here.